### DISCUSSION TITLE
⏳ New Navigator UI/UX | Concept 👨🏻‍💻

### DISCUSSION THREAD

**Discussion by Henzxon** [elementor/elementor#24922] [Feature Request]
### Prerequisites

- [X] I have searched for similar feature requests in both open and closed discussions and cannot find an existing request.
- [X] I have verified that the feature is still missing from the latest stable versions of Elementor or Elementor Pro.

### What problem is your feature request going to solve? Please describe..

I'm frustrated using the navigator and it is less helpful when managing nested containers.

- You cannot multi-select and drag multiple widgets at the same time.
- Dragging widget is a painful experience.
- Dragging nested containers is a confusing process when layout complexity increases.
- Nested widgets can only be moved one by one.

### Describe the solution you'd like.

If it is hard to improve the UX and core code and behavior of the navigator at least give a better UI. It would be great if layers or container are colored based on the context to improve navigation.

### `Old UI`
<img width="1367" alt="Original" src="https://github.com/elementor/elementor/assets/102000469/31256c15-df22-4990-9586-41ed1d490528">

### `New UI`

I created this concept in Figma to further illustrate how colors could play a huge role in terms of ease of use.

- I swapped the notes and site setting to avoid accidental click when adding widgets as it forces the page to reload.
- Global widgets are properly indicated and can be easily identified.
- Parent containers in pink are also distinguishable.
- I also moved the help icon to the bottom right corder of the screen as floating icon then moved the navigator icon to right or the screens where is should be in the first place. This is to eliminate mouse travel across the entire screen.

<img width="1367" alt="Refined" src="https://github.com/elementor/elementor/assets/102000469/d89d0dee-bd4f-4f7d-a60f-82cb27e70a8c">

> I know this may not be the best but I think this gives a better user experience.

- I also added this dedicated HTML outliner tab for accessibility which display the Semantic Structure of the page to improved the overall layout and building process.
- A visual indicator that shows nested groups and container depth properly.
- This feature can recognize incorrect html markup and highlights error automatically.  
- Instead of using a chrome extension, having it inside the builder while giving the options to adjust the markup in one panel would be awesome. Editing aria labels and attributes in one place instead of moving back and forth between widget tabs and settings.

<img width="1367" alt="New Outliner" src="https://github.com/elementor/elementor/assets/102000469/b0d68ed8-5fbc-47d8-b6c1-8bad9f32af76">

### Describe alternatives you've considered.

None

### Agreement

- [X] I agree that my request may be closed without action if it doesn't meet all the requirements.

**Comment by Muncamoo2000** [comment]
I agree with this post. One of the main things I need is the ability to move elements together. Similar to Photoshop locking layers so they can all be adjusted at once. When elements are Absolute it is really hard to move them one by one and line them up again. It would be nice to move them all together.

**Comment by 79ho3ein** [comment]
Excellent idea! Let me add three things to that.

1- Delete and duplicate buttons for each item. (These two buttons will be displayed on hover)
2- If the display condition feature and custom css are used in any of the elements, it will be visible as a simple icon.
3- Ability to lock the desired layer.

**Comment by phkoon** [comment]
One thing I miss the most is the responsive width x height indicator showing the current values. It would save a hell lot of time if it was back.
